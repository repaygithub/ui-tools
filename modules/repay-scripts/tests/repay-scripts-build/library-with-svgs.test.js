const TestSetup = require('../test-setup')

beforeEach(() => jest.setTimeout(1000 * 60 * 5))

describe('@repay/repay-scripts', () => {
  it('should build a library with svgs', async () => {
    const test = new TestSetup('library-with-svgs')
    await test.setup()

    const pkg = JSON.parse(await test.readFile('package.json'))
    await test.exec(`yarn repay-scripts build --lib index.js`)
    // relative svg import should be correct and absolute is unchanged
    expect(await test.readFile(pkg.module)).toMatchSnapshot()
  })
})
