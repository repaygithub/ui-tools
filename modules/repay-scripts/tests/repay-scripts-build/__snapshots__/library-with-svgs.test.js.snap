// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@repay/repay-scripts should build a library with svgs 1`] = `
"import internalSvg from './dist/icons/fake.svg';
import externalSvg from 'fake-svg-lib/fake.svg';

function index () {
  console.log(internalSvg);
  console.log(externalSvg);
}

export default index;
//# sourceMappingURL=build.esm.js.map
"
`;
